stac_fastapi/pgstac/stac_fastapi/pgstac/core.py

```diff
@@ -174,32 +114,32 @@ async def _search_base(
             items = await conn.fetchval(q, *p)
         next = items.pop("next", None)
         prev = items.pop("prev", None)
-        collection = ItemCollection.construct(**items)
+        collection = ItemCollection(**items)
         cleaned_features = []
-        if collection.features is None or len(collection.features) == 0:
+        if collection["features"] is None or len(collection["features"]) == 0:
             raise NotFoundError("No features found")
 
-        for feature in collection.features:
-            feature = Item.construct(**feature)
+        for feature in collection["features"]:
+            feature = Item(**feature)
```

Comment:
@geospatial-jeff here is an example of one of the places that we were using Item.construct(**feature) as opposed to Item(**feature) to avoid the pydantic overhead.

Reply:
in this case `Item` is not a pydantic model.  It's just a `TypedDict`.

Reply:
I think in this case, this is no longer a Pydantic model but a TypedDict instead (https://github.com/stac-utils/stac-fastapi/blob/custom-models/stac_fastapi/types/stac_fastapi/types/stac.py#L50), so it's creating a different type of object

Reply:
ahh, got it. missed that part.

Reply:
I'm thinking of renaming the types to `ItemDict`, `CollectionDict` etc. to be more explicit.  Too many classes named `Item` etc.